src/app/core/route-guards/auth.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+
+import { APP_ROUTE_URLS } from '@constants/app-constants';
+import { AuthService } from 'app/shared/services/auth.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+
+    if (!this.authService.getEmailAndReSendTime().email || !this.authService.getEmailAndReSendTime().reSendTime) {
```

Comment:
We should evaluate `this.authService.getEmailAndReSendTime()` one time and use it.

Reply:
done.